Sources/AsyncHTTPClient/ConnectionPool/HTTP1/HTTP1ClientChannelHandler.swift

```diff
@@ -156,6 +157,7 @@ final class HTTP1ClientChannelHandler: ChannelDuplexHandler {
             metadata: req.requestFramingMetadata
         )
         self.run(action, context: context)
+        promise?.succeed(())
```

Comment:
This doesn't seem appropriate. We should presumably send the promise on with the appropriate data, rather than silently succeeding it here.

Reply:
Good catch. I have reverted it now. I though I needed it because `try channel.writeOutbound(request)` would block indefinitely but I have now replaced it with `channel.writeAndFlush(request, promise: nil)` which doesn't block and is closer to the production code.